packages/go/dawgs/graph/node.go

```diff
@@ -251,6 +252,199 @@ func (s *NodeSet) UnmarshalJSON(input []byte) error {
 	return nil
 }
 
+type ThreadSafeNodeSet struct {
+	nodeSet NodeSet
```

Comment:
Pretty sure you can just embed `NodeSet` and `sync.RWMutex`

``` go
type ThreadSafeNodeSet struct {
    NodeSet
    sync.RWMutex
}
```

This way you can just wrap the embedded methods without having to copy-paste the implementation in most cases.

``` go
func (s ThreadSafeNodeSet) Pick() *Node {
    s.RLock()
    defer s.RUnlock()
    return s.NodeSet.Pick()
}
```